README.md

```diff
@@ -0,0 +1,580 @@
+# Xiana framework
+
+Xiana is a lightweight web-application framework written in Clojure LISP, its goal is to be simple, blasting fast, and
+most important - a welcome platform for web programmers from different backgrounds who want to experience the wonders of
+functional programming!
+
+It's easy to install, fun to experiment and a powerful tool to produce reliable code for the world wide web.
+
+## Contents
+
+- [Conventions](#conventions)
+  - [State](#state)
+  - [Monads](#monads)
+  - [Action](#action)
+  - [Handler](#handler)
+  - [Dependencies](#dependencies)
+  - [Interceptors](#interceptors)
+    - [Router and controller interceptors](#router-and-controller-interceptors)
+    - [Providing default interceptors](#providing-default-interceptors)
+    - [Typical use-case, and ordering](#typical-use-case-and-ordering)
+    - [Interceptor overriding](#interceptor-overriding)
+    - [Interceptors implemented in xiana](#interceptors-implemented-in-xiana)
+      - [acl-restrict](#acl-restrict)
+      - [log](#log)
+      - [side-effect](#side-effect)
+      - [view](#view)
+      - [params](#params)
+      - [db-access](#db-access)
+      - [message](#message)
+      - [session-user-id](#session-user-id)
+      - [session-user-role](#session-user-role)
+      - [muuntaja](#muuntaja)
+      - [session](#session)
+      - [rbac](#rbac)
+      - [coercion](#coercion)
+      - [cookies](#cookies)
+    - [Defining new interceptors](#defining-new-interceptors)
+      - [Example](#example)
+- [Modules](#modules)
+  - [Frontend](#frontend)
+  - [Backend](#backend)
+    - [Core](#core)
+    - [ACL](#acl)
+    - [Auth](#auth)
+    - [Config](#config)
+    - [Database](#database)
+    - [Core](#core)
+    - [Postgresql](#postgresql)
+    - [SQL](#sql)
+    - [Migrations](#migrations)
+    - [Interceptor](#interceptor)
+      - [Interceptor/core](#interceptorcore)
+      - [Interceptor/muuntaja](#interceptormuuntaja)
+      - [Interceptor/wrap](#interceptorwrap)
+    - [Mail](#mail)
+    - [RBAC](#rbac-1)
+    - [Route](#route)
+    - [Session](#session)
+    - [State](#state-1)
+    - [Webserver](#webserver)
+- [Usage](#usage)
+  - [From template](#from-template)
+  - [As dependency](#as-dependency)
+  - [Examples](#examples)
+- [Contribution](#contribution)
+  - [Development dependencies](#development-dependencies)
+  - [Setup](#setup)
+  - [Deps](#deps)
+  - [Leiningen](#lein)
+- [References](#references)
+
+## Conventions
```

Comment:
I think this section probably goes in the Tutorial section of the docs. It's probably most relevant for learning, right?